src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceDao.java

```diff
@@ -113,6 +113,8 @@ public GoogleBucketResource createAndLockBucket(GoogleBucketRequest bucketReques
             .addValue("name", bucketRequest.getBucketName())
             .addValue("flightid", flightId);
         DaoKeyHolder keyHolder = new DaoKeyHolder();
+        // Lock the table to avoid serialization failures during bulk load
+        jdbcTemplate.getJdbcTemplate().execute("LOCK TABLE bucket_resource IN EXCLUSIVE MODE");
```

Comment:
This is a different approach than using retries for postgres serialization errors that we've use elsewhere. This one seems more restrictive in terms of only ever allowing one insert to happen at a time, where retries should allow multiple inserts when postgres decides they don't conflict. Could you explain why you chose one approach over the other?

Reply:
I believe this is a rare case. Now that we examine the bucket state before deciding to create and lock, it is a very small window.